src/com/stanimiriliev/chaptertwo/heterogeneoustree/products/Egg.java

```diff
@@ -0,0 +1,15 @@
+package com.stanimiriliev.chaptertwo.heterogeneoustree.products;
+
+/**
+ * @author Stanimir Iliev
+ * @since 21.07.2017
+ */
+public class Egg extends Products {
```

Comment:
No, the idea is that the Product to be interface like comparable, look at the gist that I post you, there are examples with Payable and HasCode interfaces - same as your product